src/foam/u2/view/RichChoiceView.js

```diff
@@ -264,26 +266,29 @@ foam.CLASS({
           }))
           .add(this.slot(function(sections) {
             return this.E().forEach(sections, function(section) {
-              this
-                .start()
-                  .addClass(self.myClass('heading'))
-                  .add(section.heading)
-                .end()
-                .start()
-                  .select(section.filtered || section.dao, function(obj) {
-                    return this.E()
-                      .start(self.rowView, { data: obj })
-                        .enableClass('disabled', section.disabled)
-                        .callIf(! section.disabled, function() {
-                          this.on('click', () => {
-                            self.fullObject_ = obj;
-                            self.data = obj;
-                            self.isOpen_ = false;
+              var positionElement = this;
+              section.dao.select().then((resp) => {
```

Comment:
I wonder if this changing from synchronous to asynchronous is going to break the ordering for multiple sections. If there are two sections and the first section's promise takes longer to finish than the second section's promise, then I'm guessing the second section would be rendered before the first section, which we wouldn't want.

Reply:
I'm not 100% sure but I don't think that would be the case considering the select response is contained within a forEach function created by an element. @kgrgreer might be able to confirm this.

Reply:
I tested and can confirm that with these changes, the order of the sections depends on how long it takes for the count selection to resolve. I don't think it should work that way. I think the order of the sections should depend on the order that they were given in the code.

I think we should be able to use `Promise.all()` to fix this.